### DISCUSSION TITLE
A ansible playbook for upgrading semaphore... and a request to the release manager

### DISCUSSION THREAD

**Discussion by farfade** [semaphoreui/semaphore#1802] [Show and tell]
Hello, thanks for the nice work on semaphore :)

To train myself, I've written a playbook for upgrading semaphore with itself on Debian. That works ! I'm open to suggestions to improve it as it is basically my first ansible playbook !

A couple of requests to the release manager : 
- Remove the v in the folder or add a v in the filename of packages to ease automation : https://github.com/ansible-semaphore/semaphore/releases/download/**v**2.9.45/semaphore**_2**.9.45_linux_amd64.deb
- Don't change those conventions in the future

> tasks:
> 
>    - name: Check dependencies
>      ansible.builtin.apt:
>       pkg:
>        #dependency for apt deb dpkg installer
>        - xz-utils
>        #dependencies for pip (python modules manager)
>        - python3-pip
>        - virtualenv
>        - python3-setuptools
>       state: present
> 
>    - name: Install github3.py dependency for community.general.github_release
>      ansible.builtin.pip:
>       name: github3.py
>       virtualenv: /opt/ansible_venv/semaphore_upgrade/
> 
>    - name: Query github to get the latest release metadata
>      vars:
>       ansible_python_interpreter: /opt/ansible_venv/semaphore_upgrade/bin/python
>      community.general.github_release:
>       repo: semaphore
>       user: ansible-semaphore
>       action: latest_release
>      register: semaphore_latest_tag
> 
>    - name : Latest version read from github
>      ansible.builtin.debug:
>       var=semaphore_latest_tag.tag
> 
>    - name: Install the package
>      ansible.builtin.apt:
>       deb: https://github.com/ansible-semaphore/semaphore/releases/download/{{semaphore_latest_tag['tag']}}/semaphore_{{semaphore_latest_tag.tag[1:]}}_linux_amd64.deb

**Comment by bigjon440** [comment]
I think this is a great idea! One would assume that one would want to use ansible to set up your semaphore environment from scratch to enable rapid testing. Is this something that can get added into a repo so we can make contributions to it? I would be willing to work on it, although I am not an Linux expert I am try to learn more about it everyday as I love ansible but don't want the baggage with AAP or AWX.
